routes/tasks.go

```diff
@@ -0,0 +1,167 @@
+package routes
+
+import (
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jinzhu/gorm"
+	"github.com/labstack/echo"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/kirikami/go_exercise_api/database"
+)
+
+func SaveTaskHandler(c echo.Context) error {
+	const timeForm = "3 04 PM"
+	var ids int
+	var id int64
+
+	currentTime := time.Now()
+	tasks := []database.Task{}
+
+	db := c.Get("DBConnection").(*gorm.DB)
+
+	if c.FormValue("id") != "" {
+		id, err := strconv.ParseInt(c.FormValue("id"), 10, 64)
+
+		if err != nil {
+			return c.String(http.StatusInternalServerError, err.Error())
+		}
+
+		_ = id
+
+	} else {
+		err := db.Find(&tasks).Count(&ids).Error
+		if err != nil {
+			return c.String(http.StatusInternalServerError, err.Error())
+		}
+		id = int64(ids + 1)
+	}
+
+	title := c.FormValue("title")
+	description := c.FormValue("description")
+	priority, err := strconv.Atoi(c.FormValue("priority"))
+	if err != nil {
+		return c.String(http.StatusInternalServerError, err.Error())
+	}
+	createdAt := &currentTime
+	updatedAt := &currentTime
+	isDeleted := false
+	isCompeted := false
+
+	taskDocument := database.Task{id, title, description, priority, createdAt, updatedAt, nil, isDeleted, isCompeted}
+
+	err = db.Save(&taskDocument).Error
+	if err != nil {
+		return c.String(http.StatusInternalServerError, err.Error())
```

Comment:
Return json response with error details, you can use twitter or github api as an example